inf-ruby.el

```diff
@@ -683,6 +675,46 @@ This function also removes itself from `pre-command-hook'."
                 (point)))))
         (buffer-substring-no-properties (point) (line-end-position))))))
 
+(defun ruby-shell--encode-string (string)
+  "Escape all backslashes, double quotes, newlines, and # in STRING."
+  (cl-reduce (lambda (string subst)
+               (replace-regexp-in-string (car subst) (cdr subst) string))
+             '(("\\\\" . "\\\\\\\\")
```

Comment:
Also regarding the escapings here: shouldn't we also replace any single backslash found in text with a double one? E.g. when the original code already contains `\n` or `\t` or etc. Then this line would just look like:

```rb
'(("\\" . "\\\\")
```

Reply:
That's what it's doing now, you can add `(message "sending: %s" code)` to `ruby-send-string` to see what is going on easier. or ie:
```
(princ (replace-regexp-in-string "\\\\" "\\\\\\\\" "a\\nb
"))
```
It's cause backslash has to be escaped for string, and again for regex, so you end up with 4 backslash to represent a single one

Reply:
Got it. Thanks :-)